flask/app/routes.py

```diff
@@ -340,3 +355,174 @@ def get_enums():
         if issubclass(obj, Enum) and name != 'Enum':
             enums[name] = [e.value for e in getattr(models, name)]
     return jsonify(enums)
+
+
+def enum_validate(entity: db.Model, json_mixin: Dict[str, any], columns: List[str]) -> Union[None, str]:
+
+    for field in columns: 
+        if field in json_mixin:
+            column = getattr(entity, field) # the column type from the entities
+            value = json_mixin[field]
+            if hasattr(column.type, 'enums'): # check if enum
+                if value not in column.type.enums:
```

Comment:
There's one case missing here, which is if the column is nullable, in which case `None` is an acceptable value. The database transaction should also fail with a meaningful message if a nullable column is missing, so maybe ignoring None entirely might be acceptable?

Reply:
`(1048, "Column 'participant_type' cannot be null")%` in f1b718f